ui/raidboss/emulator/data/Encounter.ts

```diff
@@ -67,47 +90,39 @@ export default class Encounter {
           }
         }
       }
-      if (res && res.groups && res.groups.language)
+      if (res && res.groups && res.groups.language && isLang(res.groups.language))
         this.language = res.groups.language || this.language;
```

Comment:
And maybe something more like this?

```suggestion
      const matchedLang = res?.groups?.language;
      if (matchedLang && Util.isLang(matchedLang))
        this.language = matchedLang;
```